src/components/IndexLiquidityTab/IndexLiquiditySimulateDataTableRow.tsx

```diff
@@ -0,0 +1,191 @@
+    setIsLoading(true)
+    return await getMaxTrade(component.address, parseFloat(allowedSlippage), exchange)
+      .then((response) => {
+        setTradeError(false)
+        return response.size
+      })
+      .catch((error) => {
+        setTradeError(true)
+        console.error(error.toString())
+      })
+  }
+  const findMaxTrade = async (component: IndexComponent) => {
+    const checkExchanges = async () => Promise.all(
+      EXCHANGES.map((exchange) => checkMaxTrade(exchange, component)
+        .then((response) => {
+          return {
+            exchange,
+            response,
+            compressedResponse: response?.div(tenPowDecimals).toNumber()
+          }
+        })
+        .catch((error) => {
+          setTradeError(true)
+          console.error(error.toString())
+        })
+      )
+    ).catch((error) => {
+      setTradeError(true)
+      console.error(error.toString())
+    })
+    const resultsFromExchanges = await checkExchanges()
+    const exchangeMaxTrades = resultsFromExchanges!.map((exchangeObject) => { 
+      if (exchangeObject && exchangeObject.compressedResponse) {
+        return exchangeObject.compressedResponse
+      } 
+    })
+    const findBestTrade = () => {
+      let sorted = [...exchangeMaxTrades].sort((a,b) => a && b ? b - a : 0)
+      const indexOfBestTrade = exchangeMaxTrades.indexOf(sorted[0])
+      setIsLoading(false)
+      return resultsFromExchanges![indexOfBestTrade]
+    }
+    const bestTrade = findBestTrade()
+    setBestExchange(bestTrade!.exchange)
+    setMaxTrade(bestTrade!.response)
+  }
+  const renderDataTableRow = (component: IndexComponent | undefined) => {
+    if (!component) return null
+    const maxTradeToken =
+      maxTrade!.mul(PRICE_DECIMALS).div(tenPowDecimals).toNumber() /
+      PRICE_DECIMALS
+    const maxTradeUSD =
+      tokenPrice.mul(maxTrade!).div(tenPowDecimals).toNumber() /
+      PRICE_DECIMALS
+    const percentageChange = parseFloat(`${parseFloat(target) - parseFloat(component.percentOfSet)}`).toFixed(2);
+    const dollarChange = parseFloat(`${parseFloat(percentageChange) * 0.01 * selectedIndexMarketCap}`).toFixed(2);
+    const numberOfTrade = Math.abs(Math.round(parseFloat(`${ parseFloat(dollarChange) / maxTradeUSD }`)));
+    const estimatedCost = parseFloat(props.gasCost) * numberOfTrade * 1.3
+    console.log('no. of traders',numberOfTrade );
+    
+
+    return (
+      <>
+        <TableData>{component.symbol}</TableData>
+        <TableDataRightAlign>{component.percentOfSet}</TableDataRightAlign>
+        <TableDataRightAlign>
+          <TextField
+              value={target}
+              onChange={onTarget}
+              onBlur={(e) => findMaxTrade(component)}
+              style={textFieldStyles}
+              inputProps={{
+                autoComplete: 'new-password', // disable autocomplete and autofill
+              }}
+            />
```

Comment:
Please default Target input to 0.5% rather than 0. My bad for not specifying sooner.

Reply:
total target value must be equal to 100% so we can set the default target value to = weight%

Reply:
As the total value of target must be = 100 so we can't set it to 0.5% initially. We can set  it  equal to  %weight value

Reply:
Good idea, please do that. Thanks